sklearn/metrics/classification.py

```diff
@@ -239,6 +242,11 @@ def confusion_matrix(y_true, y_pred, labels=None):
     else:
         labels = np.asarray(labels)
 
+    if sample_weight is None:
+        sample_weight = np.ones(y_true.shape[0], dtype=np.int)
+    else:
+        sample_weight = np.asarray(sample_weight)
+
```

Comment:
nitpick: can you add a check here, to see if `sample_weight` is the same size as `y_true` and `y_pred` using `check_consistent_length`?